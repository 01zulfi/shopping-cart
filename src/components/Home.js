import React from "react";
import { useNavigate } from "react-router";
import asteroid from "../icons/asteroid.png";
import inStock from "../icons/in-stock.png";
import eco from "../icons/eco.png";
import shuttle from "../icons/shuttle.png";
import rex from "../icons/rex.png";
import triceratops from "../icons/triceratops.png";
import "../styles/Home.css";

const Home = () => {
  const navigate = useNavigate();

  const clickHandler = () => {
    navigate("/shop");
  };

  return (
    <section className="home">
      <section className="front-section">
        <div className="about-div">
          <p className="about-text-one">Shopping for some rocky goodness?</p>
          <p>Well, you've come to the right place</p>
          <button className="shop-button" onClick={clickHandler}>
            Shop now
          </button>
        </div>
        <div className="rotating-image">
          <img src={asteroid} alt="asteroid-icon" />
        </div>
      </section>
      <section className="boxes-section">
        <div className="box-one">
          <img src={inStock} alt="in-stock-icon" />
          <h3>Always in stock</h3>
          <p>
            With over 800, 000 asteroids in space, and more being discovered
            everyday, rest assured that they won't run out anytime soon. Here at
            Asteroid Store, we've handpicked a few for you to shop from.
          </p>
        </div>
        <div className="box-two">
          <img src={eco} alt="eco-icon" />
          <h3>Environment Friendly</h3>
          <p>
            With all production in space, and seemingly no waste products,
            asteroids are eco friendly. Rest assured, that your asteroid will
            not harm the environment of the tiny planet you reside in.
          </p>
        </div>
        <div className="box-three">
          <img src={shuttle} alt="shuttle-icon" />
          <h3>Free Shipping</h3>
          <p>
            All items on our store have no shipping charges. No tax, no customs,
            pay for just the big rock. Rest assured, the asteroid will be
            shipped right to your doorstep. In case of any damages, we provide
            money back guarantee.
          </p>
        </div>
      </section>
      <section className="customer-section">
        <h3>What do the customers say!?</h3>
        <div className="review-one">
          <p>
            They delivered on their promise! They claim they have the best
            asteroids but they really do. I was so surprised on these prices,
            and with no shipping? They really are the best.
          </p>
          <div className="review-author">
            <img src={rex} alt="rex-icon" />
            -T. Rex
          </div>
        </div>
        <div className="review-two">
          <p>
            Blazing upon entry, shiny before my eyes, so beautiful as the world
            would end. A magnet for the eyes, feast for hearts. Pristine, and
            certainly the most magnificent rock I've ever seen.
          </p>
          <div className="review-author">
            <img src={triceratops} alt="triceratops-icon" />
            -Triceratops
          </div>
        </div>
      </section>
      <div className="shop-button-div">
        <h1>What are you waiting for?</h1>
        <button className="shop-button-two" onClick={clickHandler}>
          Get your favorite asteroid now
        </button>
      </div>
    </section>
  );
};

export default Home;
