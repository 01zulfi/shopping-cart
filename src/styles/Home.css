section.home {
  display: flex;
  flex-direction: column;
  width: 100%;
}

section.home section {
  box-shadow: 0px 15px 30px rgb(0 0 0 / 15%);
}

section.front-section {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  margin: 25px;
  background-color: rgba(255, 255, 255, 0.07);
  border-radius: 10px;
  padding: 15px;
  gap: 5px;
}

.about-div {
  display: flex;
  width: 55%;
  flex-direction: column;
  align-items: center;
  gap: 10px;
}

.about-div p {
  font-size: 17px;
  text-align: center;
}

.about-div p.about-text-one {
  font-size: 19px;
  color: white;
  background-color: rgba(255, 255, 255, 0.09);
  border: 1px solid #d4bff9;
  padding: 9px;
  border-radius: 5px;
}

.about-div button.shop-button {
  animation: button-animation 3s infinite linear alternate;
  padding: 10px;
  border: 1px solid #d4bff9;
  color: black;
  border: 0;
  font-size: 19px;
  border-radius: 5px;
}

.rotating-image {
  flex: 1;
  padding: 15px;
  animation: rotating-image 5s linear;
  animation-iteration-count: infinite;
}

@keyframes rotating-image {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

section.boxes-section {
  display: flex;
  flex-direction: column;
  margin: 25px;
  background-color: rgba(255, 255, 255, 0.07);
  border-radius: 10px;
  padding: 15px;
  gap: 5px;
}

section.boxes-section p {
  text-align: center;
}

section.boxes-section div {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 15px;
  border-radius: 5px;
  background-color: rgba(255, 255, 255, 0.09);
  border: 1px solid #d4bff9;
}

section.boxes-section div h3 {
  margin: 5px;
}

section.boxes-section img {
  height: 65px;
}

section.customer-section {
  display: flex;
  flex-direction: column;
  margin: 25px;
  background-color: rgba(255, 255, 255, 0.07);
  border-radius: 10px;
  padding: 15px;
  gap: 5px;
}

section.customer-section h3 {
  width: 100%;
  text-align: center;
}

section.customer-section .review-one,
section.customer-section .review-two {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 15px;
  border-radius: 5px;
  background-color: rgba(255, 255, 255, 0.09);
  border: 1px solid #d4bff9;
  gap: 15px;
}

section.customer-section .review-author {
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
}

section.customer-section img {
  height: 65px;
}

.shop-button-div {
  display: flex;
  flex-direction: column;
  margin: 25px;
  background-color: rgba(255, 255, 255, 0.07);
  border-radius: 10px;
  padding: 15px;
  gap: 5px;
}

.shop-button-div h1 {
  text-align: center;
}

.shop-button-div button {
  animation: button-animation 3s infinite linear alternate;
  padding: 10px;
  border: 1px solid #d4bff9;
  color: black;
  border: 0;
  font-size: 25px;
  border-radius: 5px;
}

@keyframes button-animation {
  from {
    background-color: #bc75c9;
  }
  to {
    background-color: #ab47bc;
  }
}

.about-div button:hover {
  cursor: pointer;
  border: 1px solid white;
}

.shop-button-div button:hover {
  cursor: pointer;
  border: 1px solid white;
}

@media (min-width: 768px) {
  section.home {
    width: 75%;
  }

  section.front-section {
    height: 25%;
    margin-top: 45px;
  }

  .rotating-image {
    width: 45px;
    flex-grow: 0;
    flex-basis: 15%;
  }

  section.boxes-section {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
  }

  section.boxes-section div {
    width: 30%;
    text-align: center;
  }

  section.customer-section {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-around;
    gap: 0;
  }

  section.customer-section h3 {
    font-size: 30px;
    margin-bottom: 25px;
  }

  section.customer-section .review-one,
  section.customer-section .review-two {
    width: 45%;
  }
}
