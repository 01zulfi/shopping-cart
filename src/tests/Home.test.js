import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../components/Home";

let container = {};

beforeEach(() => {
  ({ container } = render(<Home />));
});

test("home renders", () => {
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveClass("home");
});

test("front section renders", () => {
  const frontSection = container.querySelector(".front-section");
  expect(frontSection).toBeInTheDocument();
});

test("boxes section renders", () => {
  const boxesSection = container.querySelector(".boxes-section");
  expect(boxesSection).toBeInTheDocument();
});

test("customer section renders", () => {
  const customerSection = container.querySelector(".customer-section");
  expect(customerSection).toBeInTheDocument();
});

test("shop button div renders", () => {
  const shopButtonDiv = container.querySelector(".shop-button-div");
  expect(shopButtonDiv).toBeInTheDocument();
});
