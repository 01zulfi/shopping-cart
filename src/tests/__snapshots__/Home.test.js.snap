// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home renders 1`] = `
<section
  class="home"
>
  <section
    class="front-section"
  >
    <div
      class="about-div"
    >
      <h2
        class="about-heading"
      >
        Asteroids
      </h2>
      <p
        class="about-text"
      >
        Best asteroids in town
      </p>
      <button
        class="shop-button"
      >
        Shop now
      </button>
    </div>
    <div
      class="rotating-image"
    >
      <p
        class="rotating-image-text"
      >
        Asteroid of the week
      </p>
      <img
        alt="asteroid"
        src=""
      />
    </div>
  </section>
  <section
    class="boxes-section"
  >
    <div
      class="box-one"
    >
      Always in stock
    </div>
    <div
      class="box-two"
    >
      Environment friendly
    </div>
    <div
      class="box-three"
    >
      Free shipping
    </div>
    <div
      class="box-four"
    >
      N/A
    </div>
  </section>
  <section
    class="customer-section"
  >
    <div
      class="review-one"
    >
      Amazing
    </div>
    <div
      class="review-two"
    >
      Excellent
    </div>
  </section>
  <div
    class="shop-button-div"
  >
    <button
      class="shop-button-two"
    >
      Get your favorite asteroid now
    </button>
  </div>
</section>
`;
